couchbase/resource_index.go

```diff
@@ -0,0 +1,123 @@
+package couchbase
+
+import (
+	"log"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"gopkg.in/couchbase/gocb.v1"
+)
+
+const (
+	bucketNameProperty     = "bucket_name"
+	bucketPasswordProperty = "bucket_password"
+	indexNameProperty      = "index_name"
+	indexFieldsProperty    = "index_fields"
+)
+
+func resourceIndex() *schema.Resource {
+	return &schema.Resource{
+		Create: createIndex,
+		Read:   readIndex,
+		Update: updateIndex,
+		Delete: deleteIndex,
+		Schema: map[string]*schema.Schema{
+			bucketNameProperty: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			bucketPasswordProperty: {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			indexNameProperty: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			indexFieldsProperty: {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+		},
+	}
+}
+
+func createIndex(data *schema.ResourceData, meta interface{}) (err error) {
+	manager, err := getBucketManager(data, meta)
+	if err != nil {
+		return
+	}
+	indexName := data.Get(indexNameProperty).(string)
+	indexFields := data.Get(indexFieldsProperty).(string)
+	if indexFields == "" {
+		err = manager.CreatePrimaryIndex(indexName, true, false)
+	} else {
+		err = manager.CreateIndex(indexName, strings.Split(indexFields, ","), true, false)
+	}
+	if err != nil {
+		return
+	}
+	log.Printf("[INFO] An index with the name %q was created", indexName)
+	data.SetId(indexName)
+	return
```

Comment:
It's better to call `readIndex(...)` in accordance to [documentation](https://www.terraform.io/docs/extend/writing-custom-providers.html#defining-resources):
```
The create and update function should always return the read function to ensure the state is reflected in the terraform.state file.
```